CompatBot/Commands/ForcedNicknames.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using CompatBot.Commands.Attributes;
+using CompatBot.Database;
+using CompatBot.Utils;
+using DSharpPlus.CommandsNext;
+using DSharpPlus.CommandsNext.Attributes;
+using DSharpPlus.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace CompatBot.Commands
+{
+    [Group("forced-nicknames"), RequiresWhitelistedRole]
+    [Description("Manage users who has forced nickname.")]
+    internal sealed class ForcedNicknames : BaseCommandModuleCustom
+    {
+        [Command("add")]
+        [Description("Enforces specific nickname for particular user.")]
+        public async Task Add(CommandContext ctx, 
+            [Description("Discord user to add to forced nickname list.")] DiscordMember discordMember, 
+            [Description("Nickname which should be displayed.")] string expectedNickname)
+        {
+            using (var context = new BotDb())
+            {
+                if (!(await context.ForcedNicknames.SingleOrDefaultAsync(x => x.UserId == discordMember.Id).ConfigureAwait(false) is null))
+                {
+                    await ctx.ReactWithAsync(Config.Reactions.Failure, $"{discordMember.Mention} is already on blacklist.").ConfigureAwait(false);
+                    return;
+                }
+
+                context.ForcedNicknames.Add(
+                    new ForcedNickname {UserId = discordMember.Id, Nickname = expectedNickname}
+                );
+                await context.SaveChangesAsync().ConfigureAwait(false);
+
+                await discordMember.ModifyAsync(x => x.Nickname = expectedNickname).ConfigureAwait(false);
```

Comment:
So should I then only add user to forced nicknames list with out rename and if yes, should I somehow notify the user that bot didn't have permission to rename? Or maybe should  I not allow to execute that command at all when bot doesn't have sufficient permissions?

Reply:
yes, adding it to the list would still be useful, just don't execute the modify command, or at least do not crash and burn in case something went wrong with permissions